vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -635,6 +646,238 @@ def collect_events(self):
+                    "base tags=`%s` additional tags=`%s`",
+                    metric_full_name,
+                    metric_value,
+                    hostname,
+                    base_tags,
+                    additional_tags,
+                )
+                return
+
+        tags = []  # type: List[str]
+        tags = tags + base_tags
+
+        add_additional_tags(tags, additional_tags)
+
+        # Use isEnabledFor to avoid unnecessary processing
+        if self.log.isEnabledFor(logging.DEBUG):
+            self.log.debug(
+                "Submit property metric: name=`%s`, value=`%s`, hostname=`%s`, tags=`%s`, count=`%s`",
+                metric_full_name,
+                metric_value,
+                hostname,
+                tags,
+                is_count_metric,
+            )
+        metric_method = self.count if is_count_metric else self.gauge
+        metric_method(metric_full_name, metric_value, tags=tags, hostname=hostname)
+
+    def submit_disk_property_metrics(
+        self,
+        disks,  # type: List[VmomiObject]
+        base_tags,  # type: List[str]
+        hostname,  # type: str
+        resource_metric_suffix,  # type: str
+    ):
+        # type: (...) -> None
+        for disk in disks:
+            disk_path = disk.diskPath
+            file_system_type = disk.filesystemType
+            free_space = disk.freeSpace
+            capacity = disk.capacity
+            disk_tags = {'disk_path': disk_path, 'file_system_type': file_system_type}
+
+            self.submit_property_metric(
+                'guest.disk.freeSpace',
+                free_space,
+                base_tags,
+                hostname,
+                resource_metric_suffix,
+                additional_tags=disk_tags,
+            )
+
+            self.submit_property_metric(
+                'guest.disk.capacity',
+                capacity,
+                base_tags,
+                hostname,
+                resource_metric_suffix,
+                additional_tags=disk_tags,
+            )
+
+    def submit_nic_property_metrics(
+        self,
+        nics,  # type: List[VmomiObject]
+        base_tags,  # type: List[str]
+        hostname,  # type: str
+        resource_metric_suffix,  # type: str
+    ):
+        # type: (...) -> None
+        for nic in nics:
+            device_id = nic.deviceConfigId
+            is_connected = nic.connected
+            mac_address = nic.macAddress
+            nic_tags = {'device_id': device_id, 'is_connected': is_connected, 'nic_mac_address': mac_address}
+            self.submit_property_metric(
+                'guest.net', 1, base_tags, hostname, resource_metric_suffix, additional_tags=nic_tags
+            )
+            ip_addresses = nic.ipConfig.ipAddress
+            for ip_address in ip_addresses:
+                ip_tags = deepcopy(nic_tags)
+
+                ip_tags['nic_ip_address'] = ip_address.ipAddress
```

Comment:
nit - Do you really need to `deepcopy` since you always override the same key and you recreate the dict for each `nic`?

Reply:
good point, fixed!